holopy/inference/prior.py

```diff
@@ -17,20 +17,26 @@
 # along with HoloPy.  If not, see <http://www.gnu.org/licenses/>.
 
 from copy import copy
+import operator
 
 import numpy as np
 from numpy import random
 from numbers import Number, Real
 from scipy import stats
 
 from holopy.core.metadata import get_extents, get_spacing
+from holopy.core.utils import ensure_listlike
 from holopy.core.process import center_find
 from holopy.core.holopy_object import HoloPyObject
 from holopy.scattering.errors import ParameterSpecificationError
 
 EPS = 1e-6
 
 
+def _reciprocal(val):
+    return 1 / val
+
+
 class Prior(HoloPyObject):
     """Base class for Bayesian priors in holopy."""
```

Comment:
It might be nice to add a list of methods that `Prior` subclasses are expected implement, for the future devs who may extend these. I think either adding a list in the `Prior` docstring or adding empty methods, like
```python
    def lnprob(self, p)
        raise NotImplementedError("implement in subclass")
```
works. This might be out of the scope of this PR though.

Reply:
I'm planning to write a user guide for the `Prior` class at some point, like the ones for `Scatterer` and `ScatteringTheory`. Would that be sufficient or do you think I should put something in the code as well?

Reply:
I think something in the code as well would be good. (Just because someone could update the code in the future and forget to update the user guide.) Again, not super urgent though.